wallet/src/de/schildbach/wallet/ui/dashpay/PlatformRepo.kt

```diff
@@ -239,6 +250,51 @@ class PlatformRepo(val walletApplication: WalletApplication) {
         }
     }
 
+    /**
+     *  Wraps callbacks of DeriveKeyTask as Coroutine
+     */
+    private suspend fun deriveKey(handler: Handler, wallet: Wallet, password: String): KeyParameter {
+        return suspendCoroutine { continuation ->
+            object : DeriveKeyTask(handler, walletApplication.scryptIterationsTarget()) {
+
+                override fun onSuccess(encryptionKey: KeyParameter, wasChanged: Boolean) {
+                    continuation.resume(encryptionKey)
+                }
+
+                override fun onFailure(ex: KeyCrypterException?) {
+                    //CreateIdentityService.log.error("unable to decrypt wallet", ex)
+                    continuation.resumeWithException(ex as Throwable)
+                }
+
+            }.deriveKey(wallet, password)
+        }
+    }
+
+    suspend fun sendContactRequest(userId: String, encryptionKey: KeyParameter): Resource<Nothing> {
+        return try {
+            val identity = platform.identities.get(userId)
+            println("identity: $identity")
+            val blockchainIdentityData = blockchainIdentityDataDaoAsync.load()!!
+            this.blockchainIdentity = initBlockchainIdentity(blockchainIdentityData,
+                    walletApplication.wallet)
+
+            val userId = this.blockchainIdentity.watchIdentity(3, 500, RetryDelayType.SLOW20)
```

Comment:
We get this `userId` from `this.blockchainIdentity.uniqueIdString`

What is the purpose of making a query to Platform to verify the existance of our identity?